### PR TITLE
Update convert_and_optimize_asr.py

### PR DESCRIPTION
**Pull request by zhuo-yoyowz**
Update convert_and_optimize_asr.py with quantization code of whisper model

### CODE REVIEW

**Review by adrianboguszewski — COMMENTED** [review]
Thanks, @zhuo-yoyowz. Did you test it in the app? Will it work with Optimum Intel (app uses HF interface)?

Is it possible to quantize it using OVQuantizer from Optimum Intel?

**Comment by zhuo-yoyowz** [comment]
> Thanks, @zhuo-yoyowz. Did you test it in the app? Will it work with Optimum Intel (app uses HF interface)?
> 
> Is it possible to quantize it using OVQuantizer from Optimum Intel?

Hi Adrian, I've tested it in app.py. Without changing any code in app.py, the current pipeline could also load and compile the quantized model successfully. Haven't done testing the OVQuantizer from Optimum Intel. I'm still a bit uncertain about how to se the configuration for setting up the calibration dataset and defining the preprocess function.

File: recipes/conversational_voice_agent/convert_and_optimize_asr.py [hunk 1]
```diff
@@ -1,13 +1,139 @@
...
+    original_encoder_request = ov_model.encoder.request
+    original_decoder_with_past_request = ov_model.decoder_with_past.request
+    encoder_calibration_data = []
+    decoder_calibration_data = []
+    ov_model.encoder.request = InferRequestWrapper(original_encoder_request, encoder_calibration_data)
+    ov_model.decoder_with_past.request = InferRequestWrapper(original_decoder_with_past_request,
+                                                             decoder_calibration_data)
+
+    calibration_dataset = load_dataset("librispeech_asr", "clean", split="validation", streaming=True)
+    for sample in tqdm(islice(calibration_dataset, calibration_dataset_size), desc="Collecting calibration data",
```
Comment (adrianboguszewski) [inline recipes/conversational_voice_agent/convert_and_optimize_asr.py:72]: tqdm is causing some errors for me. Let's remove it.

File: recipes/conversational_voice_agent/convert_and_optimize_asr.py [hunk 2]
```diff
@@ -16,25 +142,45 @@ def convert_asr_model(model_dir: Path) -> Path:
...
+    if not output_dir.exists():
+        ov_model = OVModelForSpeechSeq2Seq.from_pretrained(
+            MODEL_NAME, ov_config=ov_config, export=True, compile=False, load_in_8bit=False
+        )
+        ov_model.half()
+        ov_model.save_pretrained(output_dir)
+    else:
+        ov_model = OVModelForSpeechSeq2Seq.from_pretrained(
+            output_dir, ov_config=ov_config, compile=False
+        )
```
Comment (adrianboguszewski) [inline recipes/conversational_voice_agent/convert_and_optimize_asr.py:160]: I wouldn't check if the model is converted. I don't assume one will convert to FP16 first and then to INT8.

File: recipes/conversational_voice_agent/convert_and_optimize_asr.py [hunk 3]
```diff
@@ -16,25 +142,45 @@ def convert_asr_model(model_dir: Path) -> Path:
...
-    # export also processor
     asr_processor = AutoProcessor.from_pretrained(MODEL_NAME)
+
+
+    if use_quantization: 
+        # Use post-training quantization of NNCF to quantize the ASR model
+
+        CALIBRATION_DATASET_SIZE = 50
+        quantized_distil_model_path = model_dir / (MODEL_NAME.rsplit ("/")[-1] + "-INT8")
+        ov_model.to("AUTO")
```
Comment (adrianboguszewski) [inline recipes/conversational_voice_agent/convert_and_optimize_asr.py:169]: Why AUTO device here? Shouldn't be CPU or nothing?
Is compilation needed?

**Review by adrianboguszewski — CHANGES_REQUESTED** [review]
Does it work for you? If I use it in the app I don't get a meaningful transcription. Just a random word.

File: recipes/conversational_voice_agent/convert_and_optimize_asr.py (hunk 3 above)
Comment (zhuo-yoyowz) [reply recipes/conversational_voice_agent/convert_and_optimize_asr.py]: Hi Adrian, replaced codes with updated version, using Optimum-Intel for weights compression directly. Please help review. Thanks~

**Comment by adrianboguszewski** [comment]
Good job!

**Pull request merged**